### ISSUE TITLE
Trigger 'Main Kaoto UI' workflow when new commit is merged into KaotoIO/kaoto-next repository

### ISSUE THREAD

**Issue by djelinek** [KaotoIO/vscode-kaoto#490]

**Comment by djelinek** [comment]
seems we have options to use pne of existing github actions from marketplace
 - https://github.com/marketplace/actions/trigger-workflow-and-wait
 - https://github.com/marketplace/actions/workflow-dispatch

both of them will require to modify some of existing KaotoIO/kaoto-next workflow to trigger vscode-kaoto workflow at the end

**Comment by djelinek** [comment]
I can see these two candidates to add trigger vscode-kaoto workflow
- https://github.com/KaotoIO/kaoto-next/blob/main/.github/workflows/build-lint-test.yml
- https://github.com/KaotoIO/kaoto-next/blob/main/.github/workflows/deploy-main.yml

to implement in one of them should be more then anough, but which one makes more sense? may I ask for opinion @apupier @lordrip ?

**Comment by djelinek** [comment]
I would suggest to implement into `build-lint-test.yml`

the workflow could looks something like this for example 
<img width="677" alt="image" src="https://github.com/KaotoIO/vscode-kaoto/assets/18696866/3fc9dc01-9e2f-41c1-a613-6002689bfb67">

- after build ends (successfully), it will trigger appropriate pipeline in vscode-kaoto project
- it will not propagate failures, in case the vscode-kaoto pipeline would fail
- it is not waiting for finish of vscode-kaoto pipeline

**Comment by djelinek** [comment]
I have started the initial work
- kaoto-next triggering pipeline successful run - https://github.com/djelinek/kaoto-next/actions/runs/7919821856
- added one new job - https://github.com/djelinek/kaoto-next/commit/a47ed14ace206e87dab4ecdb5b432321858f9076

**Comment by lordrip** [comment]
Hi @djelinek, `build-lint-test.yml` could be a good place, but I guess it will need some editing because that workflow runs after every push to a branch, not only `main`, so I guess we wouldn't want to build VSCode in this scenarios 🤔 

There's another one, `deploy-main` which only gets executed after a push to the `main` branch, perhaps this could work better?

**Comment by djelinek** [comment]
> Hi @djelinek, `build-lint-test.yml` could be a good place, but I guess it will need some editing because that workflow runs after every push to a branch, not only `main`, so I guess we wouldn't want to build VSCode in this scenarios 🤔
> 
> There's another one, `deploy-main.yml` which only gets executed after a push to the `main` branch, perhaps this could work better?

yes, I have missed that, you are right.. `deploy-main.yml` looks as better place... or I have another option and that is provide separate workflow file `vscode-tests.yml` which will be just short trigger job run on every push into main branch.. WDYT @lordrip , is that okey for kaoto-next to provide another separate workflow for that?

**Comment by lordrip** [comment]
> yes, I have missed that, you are right.. `deploy-main.yml` looks as better place... or I have another option and that is provide separate workflow file `vscode-tests.yml` which will be just short trigger job run on every push into main branch.. WDYT @lordrip , is that okey for kaoto-next to provide another separate workflow for that?

Yup, I think that's also good, I guess that there's no need to add logic, right? it will serve just as a trigger after each push to the main branch

**Comment by djelinek** [comment]
> > yes, I have missed that, you are right.. `deploy-main.yml` looks as better place... or I have another option and that is provide separate workflow file `vscode-tests.yml` which will be just short trigger job run on every push into main branch.. WDYT @lordrip , is that okey for kaoto-next to provide another separate workflow for that?
> 
> 
> 
> Yup, I think that's also good, I guess that there's no need to add logic, right? it will serve just as a trigger after each push to the main branch

yes, exactly
